### PR TITLE
Dynamic Polling

### PR DESCRIPTION
**Pull request by mpalermo73**
This brings back an "auto" polling period rebranded as "Dynamic Polling".   The goal is to increase polling frequency when a Life360 member is moving (`inTransit`) and return to the user defined polling period when no motion is reported.

I also felt that Dynamic Polling could help with your thought of "randomizing" polling to help beat bot detectors.  WHile this isn't "randomizing", it does "randomly" change the existing polling period from time to time which may help.

I also stuck in a "3 minute" polling period option.  1 minute and less seemed overkill and 5 minutes felt too long now that we can have both overkill _and_ slow - dynamically! :wink:

Screenshot_20241222_132037-1

### CODE REVIEW

**Comment by jpage4500** [comment]
sorry i hadn't seen this until now but i'll try to test it out today

thanks!

**Comment by mpalermo73** [comment]
> sorry i hadn't seen this until now but i'll try to test it out today
> 
> thanks!

d00d it's totally OK.  I ended up needed some extra time to work out a coupler kinks, as it turned out.  There's a few different places this can get fired from, and each had their own things that needed to be worked with/around.  I can get into the upsides / downsides of the different ideas if you'd like, but this seems to be the best.

File: life360/life360_app.groovy [hunk 1]
```diff
@@ -562,3 +581,41 @@ void captureCookies(response) {
...
+    state.members.each { member ->
+    
+        // FOR TESTING DYNAMIC POLLING
+/*
+        if (member["firstName"] == "YOURNAME") {
+            state.memberInTransit = true
+        }
+*/
+
+        if (member["location"].inTransit.toInteger() == 1 && member["location"].speed.toInteger() > 1) {
```
Comment (jpage4500) [inline life360/life360_app.groovy:599]: I'm getting a NPE here (inTransit)

```
java.lang.NullPointerException: Cannot get property 'inTransit' on null object on line 599 (method handleTimerFired)
```

Comment (jpage4500) [reply life360/life360_app.groovy:599]: I pushed NPE checks to get it running; I'll check the logs after it's been running a while

Comment (mpalermo73) [reply life360/life360_app.groovy:599]: Hey!

That's interesting... I haven't noticed that here/  I just double checked on my hub and don't see that.  I've been running errands most of the afternoon and my daughter has been out 'n' about as well, so there's a lot of shifting in and out of dynamic for there to have been an error going into that function.  I'll keep an eye on it though.

Thanks for giving it a shot!

**Pull request merged**